features/steps/thens/alignment.py

```diff
@@ -352,3 +359,67 @@ def step_impl(context, inst, activation_phrase):
                         observed=observed_msg,
                         severity=OutcomeSeverity.ERROR,
                     )
+
+
+@gherkin_ifc.step('Each segment must have geometric continuity in {continuity_type}')
+def step_impl(context, inst, continuity_type):
+    """
+    Assess geometric continuity between alignment segments for ALS016, ALS017, and ALS018
+    """
+    # Ref: https://standards.buildingsmart.org/IFC/RELEASE/IFC4_3/HTML/lexical/IfcTransitionCode.htm
+    position_transition_codes = ["CONTINUOUS", "CONTSAMEGRADIENT", "CONTSAMEGRADIENTSAMECURVATURE"]
+    tangency_transition_codes = ["CONTSAMEGRADIENT", "CONTSAMEGRADIENTSAMECURVATURE"]
+
+    length_unit_scale_factor = ifcopenshell.util.unit.calculate_unit_scale(
+        ifc_file=context.model,
+        unit_type="LENGTHUNIT"
+    )
+    for pair in inst:
+        pair_continuity_was_calculated = False
+        previous, current = pair
+        entity_contexts = ifc.recurrently_get_entity_attr(context, current, 'IfcRepresentation', 'ContextOfItems')
+        precision = ifc.get_precision_from_contexts(entity_contexts)
+        precision_str = str(precision)
+        parts = precision_str.lower().split("e")
+        mantissa = parts[0]
+        sig_figs = len(mantissa)
+        if "." in mantissa:
+            sig_figs -= 1
+
+        exponent_str = parts[-1]
+        display_sig_figs = int(abs(float(exponent_str)) + sig_figs + 1)  # add one more digit for rounding confirmation
+
+        match continuity_type.lower():
+            case "position":
+                expected = precision
+                if current.Transition in position_transition_codes:
+                    observed = geometry.alignment_segment_positional_difference(
+                        length_unit_scale_factor,
+                        previous,
+                        current
+                    )
+                    pair_continuity_was_calculated = True
+
+            case "tangency":
+                if current.Transition in tangency_transition_codes:
+                    expected = math.atan2(precision, float(current.SegmentLength.wrappedValue) * 1000)
+                    observed = geometry.alignment_segment_angular_difference(
+                        length_unit_scale_factor,
+                        previous,
+                        current
+                    )
+                    pair_continuity_was_calculated = True
```

Comment:
Instead of capturing whether something was calculated which is error prone and easily forgotten I would rather explicitly handle the default case.

```suggestion
            case _: return
```

Reply:
Yes but there is an additional layer of logic regarding the `Transition` code of the segment and whether it is to be interpreted as continuous or not.  But point taken; I will revise to stack conditions with `if` statements and remove the pattern matching.

Reply:
> 1 I don't understand the factor 1000.

That remained from some debugging and shouldn't be there.  I've removed it, which requires a small tweak to a test file to ensure that it continues to fail.